packages/security/src/DefaultCredentialManager.ts

```diff
@@ -49,156 +51,224 @@ import * as keytar from "keytar"; // Used for typing purposes only
+  /**
+   * Pass-through to the superclass constructor.
+   *
+   * @param {string} service The service string to send to the superclass constructor.
+   * @param {string} displayName The display name for this credential manager to send to the superclass constructor
+   */
+  constructor(service: string, displayName: string = "Zowe Built-in Credential Manager") {
+    // Always ensure that a manager instantiates the super class, even if the
+    // constructor doesn't do anything. Who knows what things might happen in
+    // the abstract class initialization in the future.
+    super(service, displayName);
+  }
+
+  /**
+   * Called by {@link CredentialManagerFactory.initialize} before the freeze of the object. This
+   * gives us a chance to load keytar into the class before we are locked down. If a load failure
+   * occurs, we will store the error and throw it once a method of this class tries to execute. This
+   * prevents a missing keytar module from stopping all operation of the cli.
+   *
+   * In the future, we could go even further to have keytar load into a sub-object of this class so
+   * that the load doesn't hold up the main class execution.
+   *
+   * @returns {Promise<void>} A promise that the function has completed.
+   */
+  public async initialize(): Promise<void> {
+      try {
+        // tslint:disable-next-line:no-implicit-dependencies
+        this.keytar = await import("keytar");
+      } catch (error) {
+          this.loadError = new ImperativeError({
+              msg: "Keytar not Installed",
+              causeErrors: error
+          });
+      }
+  }
+
+  /**
+   * Calls the keytar deletePassword service with {@link DefaultCredentialManager#service} and the
+   * account passed to the function by Imperative.
+   *
+   * @param {string} account The account for which to delete the password
+   *
+   * @returns {Promise<void>} A promise that the function has completed.
+   *
+   * @throws {@link ImperativeError} if keytar is not defined.
+   * @throws {@link ImperativeError} when keytar.deletePassword returns false.
+   */
+  protected async deleteCredentials(account: string): Promise<void> {
+    this.checkForKeytar();
+
+    if (!await this.deleteCredentialsHelper(account)) {
+      throw new ImperativeError({
+        msg: "Unable to delete credentials.",
+        additionalDetails: this.getMissingEntryMessage(account)
+      });
     }
+  }
+
+  /**
+   * Calls the keytar getPassword service with {@link DefaultCredentialManager#service} and the
+   * account passed to the function by Imperative.
+   *
+   * @param {string} account The account for which to get credentials
+   * @returns {Promise<SecureCredential>} A promise containing the credentials stored in keytar.
+   *
+   * @throws {@link ImperativeError} if keytar is not defined.
+   * @throws {@link ImperativeError} when keytar.getPassword returns null or undefined.
+   */
+  protected async loadCredentials(account: string): Promise<SecureCredential> {
```

Comment:
The `optional` argument should be added back here. I believe removing it is a breaking change.

Without this argument, CLI commands fail if an optional secure field (such as "secure_config_props") cannot be loaded. (An example of when this happens is in an Ubuntu Docker container, if Keytar fails to load and zowe.config.json has a non-empty "secure" array.)

This commit shows the behavior of the `optional` argument when it was added: https://github.com/zowe/zowe-cli-scs-plugin/pull/11/commits/f35697d8baddfa78bafe8c06cef0e42afc96da39

Reply:
I copied logic from the correct, newer, SCS repo. It contained the logic that you identified. I also altered some items that we concluded in team meetings. If you see anything else that was added in the past which I still may have overlooked, let me know.